noether/src/segmentation_server.cpp

```diff
@@ -55,28 +57,36 @@ class SegmentationAction
     bool laplacian_edge_smoothing = cfg.laplacian_edge_smoothing;
     bool laplacian_boundary_smoothing = cfg.laplacian_boundary_smoothing;
 
-
     // Set defaults if invalid data given (or not set). Note: No guarantee of fitness of defaults
     curvature_threshold = (curvature_threshold < 0.0001) ? 0.05 : curvature_threshold;
-    min_cluster_size = (min_cluster_size ==0) ? 500 : min_cluster_size;
+    min_cluster_size = (min_cluster_size == 0) ? 500 : min_cluster_size;
     max_cluster_size = (max_cluster_size == 0) ? 1000000 : max_cluster_size;
+    neighborhood_size = (neighborhood_size < 0.0001) ? 0.05 : neighborhood_size;
 
     windowed_sinc_iterations = (windowed_sinc_iterations == 0) ? 20 : windowed_sinc_iterations;
-    windowed_sinc_pass_band = (windowed_sinc_pass_band < 0.0001) ?  0.1 : windowed_sinc_pass_band;
+    windowed_sinc_pass_band = (windowed_sinc_pass_band < 0.0001) ? 0.1 : windowed_sinc_pass_band;
```

Comment:
I'd prefer to print/throw en error when a parameter value is out of range rather the overriding it with a default without the user knowing.  But this is probably outside the scope of this PR

Reply:
These were mainly to catch values that were not set to apply defaults, not so much error checking. I just did the small value since they are doubles. But yes, a warning might be nice.